oeps/oep-0013.rst

```diff
@@ -0,0 +1,576 @@
+    - See the note below for more information on ``PATCH``.
+
+- ``PUT`` - Replace a resource in entirety.  
+
+    - Most use cases can be solved using partial updates via ``PATCH``
+    - ``PUT`` can be used if a complete update of the resource, including all of its sub-resources, is desired. 
+    - A typical ``PUT`` use case:  The resource is a singular primitive value addressable via URI, such as when updating the value for a particular User Preference.
+
+- ``DELETE`` - Remove a resource (or a relationship) from the system
+
+    - **Must** return ``HTTP 204 No Content`` ::
+
+        DELETE /users/:id
+        DELETE /users/:id/groups/:group_id
+
+**A note on HTTP PATCH**
+
+"Plain" HTTP PATCH (RFC 5789) is neutral with respect to content type and only specifies that the request body provide instructions for how to update the resource in question, not the format of those instructions.  Several flavors of PATCH specific to JSON documents have more explicit definition, as noted above.
+
+
+In DRF, the default PATCH handling fairly closely resembles what is specified by merge patch, but it does not require (or understand) the ``application/merge-patch+json`` media type.  In edX REST APIs, if an implementation will use DRF's default PATCH handling, the implementation MUST recognize and accept the merge patch media type; API clients SHOULD use this media type, preferring it to the more typical ``application/json`` type.
+
+The reason for this stricture is to explicitly leave room for supporting multiple PATCH styles in any given API, should this become desirable, without breaking existing clients.
+
+URL parameters
+--------------
+
+- *Must*: Sweep complexity behind the ``?`` ::
+
+    GET /dogs?color=red&state=running&location=park
+
+- ``fields`` parameter - Allow clients to specify/filter the fields in the response by supporting a fields parameter as a comma-delimited list. ::
+
+    /dogs?fields=name,color,location
+    /dogs?fields=title,media:group(media:thumbnail)
+
+- ``expand`` parameter - Allow clients to request including data from other resources using the expand parameter as explained further in the Expansion section.
+
+.. todo: link to expansion section 
+
+- ``requested_fields`` parameter - Allow clients to request additional optional fields that will be added to the response. ::
+
+    GET /comment?requested_fields=author_profile_image,endorse_profile_image
+
+- ``text_search`` parameter - Allow clients to perform text search
+
+    global: ``/search?text_search=fluffy+fur``
+
+    scoped: ``/owners/5678/dogs?text_search=fluffy+fur``
+
+    formatted: ``/search.xml?text_search=fluffy+fur``
+
+Errors
+------
+
+**HTTP status codes**
+
+Use the appropriate value amongst the following (or document in the code why an exception is needed)
+
+    200 - OK
```

Comment:
Nit: this section shows up as quoted on the rendered page:

https://github.com/edx/open-edx-proposals/blob/6abc23aed18fe98f0270934363d744fc3ef2cdec/oeps/oep-0013.rst

I'd probably also bold the status codes themselves.